rb_ws/src/buggy/scripts/auton/autonsystem.py

```diff
@@ -115,18 +116,33 @@ def update_other_steering_angle(self, msg):
         with self.lock:
             self.other_steering = msg.data
 
+    def update_rtk_status(self, msg):
+        with self.lock:
+            self.rtk_status = msg.data
+
+    def init_check(self):
+        # checks that messages are being receieved
+        # (from both buggies if relevant),
+        # RTK status is fixed
+        # covariance is less than 1 meter
+        if (self.self_odom_msg == None) or (self.has_other_buggy and self.other_odom_msg == None) or (self.rtk_status <= 5) or (self.self_odom_msg.pose.covariance[0] ** 2 + self.self_odom_msg.pose.covariance[7] ** 2 > 100):
+            return False
+
+        # waits until rtk is fixed and covariance is normal to check heading
+        if (abs(self.cur_traj.get_heading_by_distance(start_dist)) > np.pi/2):
```

Comment:
"covariance is acceptable"

Are we trying to check if the heading seems reasonable here? If so, this may be problematic in the sim when we spawn the buggy in locations that isn't at the start/facing the "correct" direction. I'm a little confused what this is trying to check.

Reply:
We should check abs(get_heading_by_distance - current_heading) < epsilon.

Since the heading will flip 180 degrees in case of rollback, the current heading should be accepted as long as the difference is less than 180 degrees (plus some margin of error). Imo a good value for epsilon is 90 degrees. This way, the sim can start as long as the starting pose is in the general direction of the course.

Reply:
I agree, this makes more sense.